main/src/ca/uwaterloo/flix/api/lsp/provider/PhasesProvider.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp.provider
+
+object PhasesProvider {
+  val phases: List[String] = List(
+    "Parser",
```

Comment:
A smart trick is to do the following:

```
classOf[Typer].getSimpleName
```

since this is robust to refactorings and deletions.

Reply:
It also works for search for uses.

Reply:
It does seem to work for `object` and javas `Typer.getClass.getSimpleName` appends a `$` at the end.

Additionally, it seems that Parser is not accessible for some reason?

Reply:
Are these Scala objects? Maybe you need `classOf[Typer.type` or so?

Reply:
I think objects just end with `$` in their name. If so, you could just strip that.